src/test/java/us/kbase/test/auth/client/internal/TokenCacheTest.java

```diff
@@ -0,0 +1,125 @@
+package us.kbase.test.auth.client.internal;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+
+import java.lang.reflect.Field;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.junit.Test;
+
+import us.kbase.auth.AuthToken;
+import us.kbase.auth.client.internal.TokenCache;
+import us.kbase.test.common.TestCommon;
+
+public class TokenCacheTest {
+	
+	// These tests are mostly copied from the old auth repo
+	
+	private static final List<AuthToken> TEST_TOKENS = IntStream.range(1, 6).
+			mapToObj(i -> new AuthToken("token" + i, "user" + i)).collect(Collectors.toList());
+
+	@Test
+	public void constructFail() throws Exception {
+		failConstruct(0, 1, "size and maxsize must be > 0");
+		failConstruct(1, 0, "size and maxsize must be > 0");
+		failConstruct(2, 1, "size must be < maxsize");
+	}
```

Comment:
might want to add one to test for size == maxsize

Reply:
Done